libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2521,6 +2521,12 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 	annotation.referencedDeclaration = possibleMembers.front().declaration;
 	annotation.type = possibleMembers.front().type;
 
+	if (dynamic_cast<FunctionType const*>(exprType) && (memberName == "value" || memberName == "gas"))
+		m_errorReporter.warning(
+			_memberAccess.location(),
+			"Applying \"." + memberName + "\" to functions is deprecated. Use \"{" + memberName + ": }\" instead."
```

Comment:
We just discussed the wording a bit - since it's not always strictly speaking a function, the least controversial message we could come up with is:
```suggestion
			"Using \"." + memberName + "(...)\" is deprecated. Use \"{" + memberName + ": ...}\" instead."
```
But feel free to make a better suggestion :-).